server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.server.account;
+
+import io.spine.examples.chatspn.AccountCreationId;
+import io.spine.examples.chatspn.account.ReservedEmail;
+import io.spine.examples.chatspn.account.User;
+import io.spine.examples.chatspn.account.UserProfile;
+import io.spine.examples.chatspn.account.command.CreateAccount;
+import io.spine.examples.chatspn.account.event.AccountCreated;
+import io.spine.examples.chatspn.account.event.AccountNotCreated;
+import io.spine.examples.chatspn.account.event.EmailReserved;
+import io.spine.examples.chatspn.account.event.UserRegistered;
+import io.spine.examples.chatspn.account.rejection.ReservedEmailRejections.EmailAlreadyReserved;
+import io.spine.examples.chatspn.server.ChatsContext;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.core.given.GivenUserId;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.chatspn.server.given.AccountCreationTestEnv.sendRandomCreateAccountCommand;
+import static io.spine.testing.TestValues.randomString;
+
+@DisplayName("`AccountCreation` should")
+class AccountCreationTest extends ContextAwareTest {
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("emit the `AccountCreated` event if the process is finished successfully and archive itself")
+    void createdEvent() {
+        CreateAccount command = sendRandomCreateAccountCommand(context());
+        AccountCreated expectedEvent = AccountCreated
```

Comment:
Could you please start extracting some of code devoted to signal creation, to the respective test environment?